tests/test_eigen_tensor.py

```diff
@@ -0,0 +1,252 @@
+import sys
+
+import pytest
+
+np = pytest.importorskip("numpy")
+eigen_tensor = pytest.importorskip("pybind11_tests.eigen_tensor")
+submodules = [eigen_tensor.c_style, eigen_tensor.f_style]
+try:
+    from pybind11_tests import eigen_tensor_avoid_stl_array as avoid
+
+    submodules += [avoid.c_style, avoid.f_style]
+except ImportError:
+    pass
+
```

Comment:
One minor nit is that its' probably not the best to have all the tests use the same tensor_ref in case they mutate the state and cause subsequent tests to fail. It's usually best to have a func `get_tensor_ref()` that constructs the for each test or better yet have it be a pytest fixture. Not sure if eigen_matrix does this, but it's really not best practice.

Reply:
Done